### ISSUE TITLE
media delay 2-5s

### ISSUE THREAD

**Issue by ifzm** [signalwire/freeswitch#654]
FreeSWITCH Version：1.10.2~64bit ( 64bit)
OS：Windos10/Centos7.6

I used a SIP phone to call the SIP.js webpage. After repeated tests, it was confirmed that the voice could not be transmitted before 2-5s after connection, and the call was normal afterwards.

Hope to get help, here is a key log

```log
2020-05-24 21:56:54.152240 [NOTICE] sofia.c:7412 Ring-Ready sofia/internal/1001@192.168.0.104:39596!
2020-05-24 21:57:02.012396 [NOTICE] sofia.c:8479 Channel [sofia/internal/1001@192.168.0.104:39596] has been answered
2020-05-24 21:57:03.352053 [NOTICE] switch_rtp.c:1293 Auto Changing audio stun/rtp/dtls port from 172.17.19.129:63780 to 192.168.0.104:63781 idx:1
2020-05-24 21:57:05.172398 [INFO] switch_rtp.c:3273 Changing audio DTLS state from HANDSHAKE to SETUP
2020-05-24 21:57:05.172398 [INFO] switch_rtp.c:3180 audio Fingerprint Verified.
2020-05-24 21:57:05.172398 [INFO] switch_rtp.c:4212 Activating audio Secure RTP SEND
2020-05-24 21:57:05.172398 [INFO] switch_rtp.c:4190 Activating audio Secure RTP RECV
2020-05-24 21:57:05.172398 [INFO] switch_rtp.c:3222 Changing audio DTLS state from SETUP to READY
2020-05-24 21:57:05.192368 [NOTICE] switch_ivr_originate.c:3751 Channel [sofia/internal/1000@192.168.0.104] has been answered
2020-05-24 21:57:05.212367 [NOTICE] switch_core_media.c:15856 Activating write resampler
2020-05-24 21:57:05.292473 [NOTICE] switch_core_media.c:15856 Activating write resampler
2020-05-24 21:57:07.672007 [NOTICE] sofia.c:1089 Hangup sofia/internal/1001@192.168.0.104:39596 [CS_EXCHANGE_MEDIA] [NORMAL_CLEARING]
```

It is observed that the voice can only be heard after Activating write resampler appears

**Comment by altair86** [comment]
That's actual for me too

**Comment by ifzm** [comment]
I tried the following answers and the sound delay problem was successfully solved, but it is not clear if there are other effects :(
https://www.programmersought.com/article/71587077533/

**Comment by areski** [comment]
@ifzm  It might be good to leave the issue open, so that a fix can be added/merged into FreeSWITCH

**Comment by mjerris** [comment]
> I tried the following answers and the sound delay problem was successfully solved, but it is not clear if there are other effects :(
> https://www.programmersought.com/article/71587077533/

for sure doing this will break some calls that would work properly without this change.  Your real issue is with your client configuration

**Comment by mjerris** [comment]
> @ifzm It might be good to leave the issue open, so that a fix can be added/merged into FreeSWITCH

the 2-5 second delay is for our fallback way of making a call still work when a client is misconfigured.  the correct fix is on the client side, the fix he references shouldn’t be used and will break valid cases from working

**Comment by ifzm** [comment]
@mjerris thank you for pointing out the problem, I think I may not be able to confirm where the problem is on the client side, because the same configuration, in version 1.4.x the call is not delayed, but in 1.10.x it appears and after trying still not solved, so I found this reference, do you have any experience in this area to point out? Thank you very much! )

**Comment by andywolk** [comment]
@ifzm See if that helps: https://github.com/signalwire/freeswitch/pull/1284

**Comment by andywolk** [comment]
@ifzm Any updates?

**Comment by andywolk** [comment]
@altair86 Can you give #1284 a try?

**Comment by ifzm** [comment]
@andywolk Thank you for the PR case, I have not had time to test it recently, I will find time to test it in the near future to see if it works, thanks )

**Comment by andywolk** [comment]
@altair86 @ifzm Ping.

**Comment by altair86** [comment]
@andywolk hello.
I make mistake in my Freeswitch.
First change in Freeswitch:
` <param name="apply-candidate-acl" value="wan.auto"/> `

Second I setup STUN server on server with Freeswitch, and change link in js-client. And connection to media takes 0,1-0,7 sec

**Comment by andywolk** [comment]
@altair86 This is with the PR or without? Are you able to test the PR?

**Comment by altair86** [comment]
@andywolk what is "PR" mean? How to test it?

**Comment by macrocreation** [comment]
> 
> 
> > @ifzm It might be good to leave the issue open, so that a fix can be added/merged into FreeSWITCH
> 
> the 2-5 second delay is for our fallback way of making a call still work when a client is misconfigured. the correct fix is on the client side, the fix he references shouldn’t be used and will break valid cases from working

What fix can happen on the client side to solve this?

**Comment by pedroCore** [comment]
I think this is fixed in the release 1.10.7. Check it out!

**Comment by DustinQi** [comment]
> I think this is fixed in the release 1.10.7. Check it out!

Yes! I updated my freeswitch to 1.10.7,  then the media delay gone !!!

**Comment by IOMurphy** [comment]
您好！邮件我已收到，我会尽快阅读并且给您回复。
